lib/mix/test/mix/cli_test.exs

```diff
@@ -60,6 +60,13 @@ defmodule Mix.CLITest do
     end
   end
 
+  test "tasks with slashes in them raise a NoTaskError right away" do
+    in_fixture "no_mixfile", fn ->
+      contents = mix ~w[my/task]
+      assert contents =~ "** (Mix) The task my/task could not be found"
```

Comment:
@josevalim oooops, sorry I don't know how or why but I didn't see this comment, that's why I didn't fix anything for all this time. Thanks for merging and taking care of it!